gpt_json/types_oai.py

```diff
@@ -0,0 +1,60 @@
+from typing import Dict, List, Optional, Union
+
+
+class ChatCompletionDelta:
+    def __init__(
+        self,
+        content: str | None,
+        role: str | None
+    ):
+        self.content = content
+        self.role = role
+
+class ChatCompletionChunkChoice:
+    def __init__(
+        self,
+        delta: Dict[str, str],
+        finish_reason: Optional[str],
+        index: int,
+    ):
+        self.delta = delta
+        self.finish_reason = finish_reason
+        self.index = index
+
+class ChatCompletionChunk:
+    def __init__(
+        self,
+        choices: List[ChatCompletionChunkChoice],
+        created: int,
+        id: str,
+        model: str,
+        object: str,
+    ):
+        self.choices = choices
+        self.created = created
+        self.id = id
+        self.model = model
+        self.object = object
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Union[List[Dict[str, Union[Dict[str, str], None, int]]], int, str]]) -> "ChatCompletionChunk":
```

Comment:
Given this function's complexity this seems better as a BaseModel.

It looks like you're copying the fields 1:1 from the JSON payload coming back from OpenAI, so you might just be able to make use of BaseModel's default from_dict implementation to handle the verification of parameters and nested types for you?

Reply:
omg that's so much better lol